src/routes/formsg/formsgSiteLaunch.ts

```diff
@@ -225,23 +227,91 @@ export class FormsgSiteLaunchRouter {
     await mailer.sendMail(requesterEmail, subject, html)
   }
 
-  digDomainRecords = async (
-    domain: string,
-    digType: DigType
-  ): Promise<DigResponse | null> =>
-    dig([domain, digType])
-      .then((result: DigResponse) => {
-        logger.info(`Received DIG response: ${JSON.stringify(result)}`)
+  digAAAADomainRecords = async (
+    launchResult: SiteLaunchResult
+  ): Promise<DigDNSRecord[]> => {
+    try {
+      // check for AAAA records
+      const quadADigResponses = await dnsPromises.resolve6(
+        launchResult.primaryDomainSource
+      )
+
+      if (quadADigResponses.length <= 0) {
+        return []
+      }
+      return quadADigResponses.map((record) => ({
+        domain: launchResult.primaryDomainSource,
+        type: "AAAA",
+        value: record,
+      }))
+    } catch (e: any) {
+      if (e.code && e.code === "ENODATA") {
+        logger.info(
+          `Unable to get dig response for domain: ${launchResult.primaryDomainSource}. Skipping check for AAAA records`
+        )
+      }
+      throw e
+    }
+  }
+
+  digCAADomainRecords = async (
+    launchResult: SiteLaunchResult
+  ): Promise<{
+    addAWSACMCertCAA: boolean
+    addLetsEncryptCAA: boolean
+  }> => {
+    try {
+      const caaDigResponses: CaaRecord[] = await dnsPromises.resolveCaa(
+        launchResult.primaryDomainSource
+      )
+
+      if (caaDigResponses.length > 0) {
+        const caaRecords = caaDigResponses
```

Comment:
should we flip it so the default case is up top?

Reply:
[5e7d5e9](https://github.com/isomerpages/isomercms-backend/pull/1246/commits/5e7d5e9ca075851f0042b7bfc43852c8812bcf53)